File: src/Services/API/Model/Procedure/ProcedureMetaData.cs
namespace API.Model.Procedure
{
    public class ProcedureMetadata
    {
        public ExecutionPlan executionPlan;
    }

    public class CreateScript
    {
        public string Description { get; set; } // Fixed spelling
    }

    public class Dependencies
    {
        public string ReferencingObjectName { get; set; } // PascalCase for C#
        public string ReferencingObjectType { get; set; }
        public string ReferencedObjectName { get; set; }
    }

    public class ExecutionPlan
    {
        public string QueryPlanXml { get; set; } // PascalCase for XML
        public string UseAccounts { get; set; }
        public string CacheObjectType { get; set; }
        public string SizeInBytes { get; set; }  // More descriptive
        public string SqlText { get; set; }
    }

    public class Parameters
    {
        public int Id { get; set; }
        public bool HideEdit { get; set; } // Boolean type
        public string ParameterName { get; set; }
        public string Type { get; set; }
        public string Length { get; set; }
        public string Precision { get; set; } // Full word for clarity
        public string Scale { get; set; }
        public string ParameterOrder { get; set; } // More descriptive
        public string ExtendedProperty { get; set; } // Corrected spelling
    }

    public class ExtendedProperties
    {
        public string Description { get; set; } // Fixed spelling
    }

}

File: src/WebClient/mssql.client/src/app/procedure/model/Procedure-metadata.ts
export class ProcedureMetadata {
  // Add properties if necessary
}

export class CreateScript {
  public description: string; // Fixed spelling
}

export class Dependencies {
  public referencingObjectName: string; // Used camelCase for consistency
  public referencingObjectType: string;
  public referencedObjectName: string;
}

export class ExecutionPlan {
  public queryPlanXml: string; // Used proper camelCase for XML
  public useAccounts: string;  // Used camelCase for consistency
  public cacheObjectType: string;
  public sizeInBytes: string;  // Renamed to be more descriptive and accurate
  public sqlText: string;
}

export class Parameters {
  public id: number;
  public hideEdit: boolean; // Used camelCase
  public parameterName: string; // Used camelCase
  public type: string; // Simple name, more readable
  public length: string;
  public precision: string; // Used full word for clarity
  public scale: string;
  public parameterOrder: string; // More descriptive
  public extendedProperty: string; // Corrected spelling
}

export class ExtendedProperties {
  public description: string; // Fixed spelling
}
